dsn.go

```diff
@@ -51,6 +53,194 @@ type Config struct {
 	Strict                  bool // Return warnings as errors
 }
 
+// FormatDSN formats the given Config into a DSN string which can be passed to
+// the driver.
+func (cfg *Config) FormatDSN() string {
+	var buf bytes.Buffer
+
```

Comment:
The boilerplate code here is minimal and is actually useful as it allows to save a few allocations and writes in some cases.